library/alloc/src/boxed.rs

```diff
@@ -2036,6 +2038,42 @@ impl<I> FromIterator<I> for Box<[I]> {
     }
 }
 
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<I, A: Allocator> !Iterator for Box<[I], A> {}
+
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<I, A: Allocator> IntoIterator for Box<[I], A> {
+    type IntoIter = vec::IntoIter<I, A>;
+    type Item = I;
+    fn into_iter(self) -> vec::IntoIter<I, A> {
+        self.into_vec().into_iter()
+    }
+}
+
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<'a, I, A: Allocator> !Iterator for &'a Box<[I], A> {}
```

Comment:
why is this needed? is the `!Iterator` impl for `[_]` not enough?

Reply:
Unfortunately not. :(

Reply:
A comment here for why they're needed would be very 🆒